### ISSUE TITLE
Case-Sensitivity For Windows Drive Letters

### ISSUE THREAD

**Issue by jnhyatt** [gulrak/filesystem#70]
Windows case-sensitivity for drive letters is not always interpreted correctly.

To Reproduce:
```c++
fs::path path = fs::relative("c:\\dev\\working\\directory\\file.txt");
```
Assuming the current working directory is `C:\dev\working\directory`, one would expect `path` to be `file.txt`. However, because the second (optional) argument to `relative` is evaluated as `C:\dev\working\directory` (capital 'C'), `lexically_relative` will exit early on its first check and simply return a default-constructed `path`.

A fix for this specific issue could be to force drive letters to be capital, say, in `weakly_canonical`. Now, even Windows can't guarantee case-insensitivity for all file names, but I do believe that drive letters are case-insensitive -- for example, I don't think you can have a `d:` and `D:` drive mounted simultaneously. Additionally, under MSVC, it appears that path equality checks are completely case-insensitive when using the standard template library's file system implementation. This does not seem to be the case when compiling under any other compiler.

**Comment by gulrak** [comment]
Good catch, I'll look into it.

**Comment by gulrak** [comment]
It took me some time to decide how to tackle this. In the end, my take is to make the comparison for the root name on Windows (drive letter) case insensitive. As this is in part correlating with the different `fs::path::compare()` rules from [LWG 2936](https://cplusplus.github.io/LWG/issue2936), I started implementing an option to enable that one as well, but the drive letter case insensitivity is not part of that option, so should be active starting from v1.3.6 if there are no objections with that solution.

I plan to keep `LWG_2936_BEHAVIOUR` disabled by default in v1.3.6, but I either enable it by default on v1.4.0 or at least when C++20 is found.

**Comment by gulrak** [comment]
[Released with v1.3.6](https://github.com/gulrak/filesystem/releases/tag/v1.3.6)
